docs/smart-contracts/advanced/debug-cbor.md

```diff
@@ -0,0 +1,412 @@
+
+### Redeemers
+
+Field 5 is the list of redeemers. Each redeemer refers to the execution of a Plutus script.
+
+The CDDL specifies that a [redeemer](https://github.com/IntersectMBO/cardano-ledger/blob/c2b7ea777317dd1dfeba576d044be2cbe742d9a8/eras/conway/impl/cddl-files/conway.cddl#L677) is as follows:
+
+```rust
+redeemers = 
+  [ + [ tag      : redeemer_tag
+  , index    : uint .size 4
+  , data     : plutus_data 
+  , ex_units : ex_units    
+  ]
+```
+
+So, a redeemer is a 4-uple with the following components:
+
+- tag: It specifies the type of redeemer, and has four possible values. Value 0 is used for spending a script UTxO, value 1 for minting/burning with a Plutus minting policy. Don’t worry about values 2 and 3 as they has to do with staking.
+
+- index: It is an integer with a different meaning depending on the tag. For the spending tag (value 0), the index refers to the position in the inputs list after ordering it lexicographically according to the TxId and TxIdx. For the minting tag (value 1), the index refers to the position in the lexicographically ordered list of policy IDs present in the minting field.
+
+- data: This is arbitrary data that is passed as a parameter to the script. Most times this data is what is actually called the “redeemer”, instead of the complete 4-uple.
+
+- ex_units: The budget for the script execution in memory and CPU units. These numbers are used to compute the fee and must be higher or equal to the actual units used by the script execution. Execution units are computed according to the cost model, part of the protocol parameters of the Cardano blockchain. There is also a limit of the total memory and CPU units that all redeemers of a transaction can use. Also defined in the protocol parameters, Alonzo started with limits of 10.000 million units for CPU, and 10 million units for memory. However it was soon noticed that the memory limit was too low and was later raised to 16 million units. Execution units and their limits are a big issue in Cardano, as they impose important restrictions to smart contracts and developers must pay special attention to on-chain code optimization.
+
+In our example, we have only one redeemer for spending the script UTxO that is the first input according to the lexicographic order:
+
+```rust
+5: [[0, 0, 121([]), [1302238, 360901332]]]
+```
+
+We use 121([]) as the redeemer data to indicate to the script that we are trying to perform an increment operation. If it was 122([]), it would be a decrement operation. The script will validate, among other things, that the datum is updated according to the operation. The execution budget is 1302238 memory units and 360901332 CPU units, and was obtained in the balancing stage by doing an off-chain run of the validation.
```

Comment:
```suggestion
We use `121([])` as the redeemer data to indicate to the script that we are trying to perform an increment operation. If it was `122([])`, it would be a decrement operation. The script will validate, among other things, that the datum is updated according to the operation. The execution budget is 1302238 memory units and 360901332 CPU units, and was obtained in the balancing stage by doing an off-chain run of the validation.
```
Likewise these expressions also need to be quoted (as with every bit of CDDL that isn't a simple integer) so they match what's in the code blocks... I think these were also missed inadvertently since other expressions are properly quoted & so could use a further check to ensure it's done consistently.